tests/samplers_tests/test_grid.py

```diff
@@ -215,10 +215,10 @@ def test_same_seed_trials() -> None:
 
 def test_reseed_rng() -> None:
```

Comment:
I prefer to move all `test_reseed_rng` of samplers into `tests/samplers_tests/test_samplers.py` and use `parametrize`. Is it reasonable?

Reply:
Thanks. I'll create a parameterised version to compare which approach is better.

Reply:
What do you think about this unified implementation?
https://github.com/optuna/optuna/pull/3569/commits/53a903239267dbb3de6629805d7d3b22b20c8779